sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -2374,6 +2408,13 @@ private void load(
               parseStatus(jobService.pollJob(jobRef, Bound.LOAD_JOB_POLL_MAX_RETRIES));
           switch (jobStatus) {
             case SUCCEEDED:
+              if (tableDescription != null) {
+                datasetService.patchTableDescription(
+                    projectId,
+                    ref.getDatasetId(),
+                    ref.getTableId(),
+                    tableDescription);
+              }
```

Comment:
you probably want this to run after copy job too, because for a very large table we load to temp tables then copy to the final table.